wpiutil/src/main/native/thirdparty/llvm/include/wpi/StringExtras.h

```diff
@@ -538,6 +540,52 @@ constexpr std::pair<std::string_view, std::string_view> rsplit(
           slice(str, idx + separator.size(), std::string_view::npos)};
 }
 
+/**
+ * Splits @p str into substrings around the occurrences of a separator string.
+ *
+ * Each substring is passed to the callback @p func. If @p maxSplit is >= 0, at
+ * most @p maxSplit splits are done and consequently <= @p maxSplit + 1
+ * elements are added to arr.
+ * If @p keepEmpty is false, empty strings are not included. They
+ * still count when considering @p maxSplit
+ * An useful invariant is that
+ * separator.join(all callbacks) == str if maxSplit == -1 and keepEmpty == true
```

Comment:
```suggestion
 * separator.join(all callbacks) == str if keepEmpty == true.
```
Even if `maxSplit` restrains the number of splits, this will still be true because the last separators will still be contained in the last callback. (`split("1,2,3", ",", 1, true, []{})` calls the callback with "1" and "2,3", which still join to form "1,2,3".)

Reply:
This comment is the same as the non-SmallVector version, which came from LLVM, so I don't know?